src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,48 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exceptions.DuplicateAnExistingLoginException;
+import core.basesyntax.exceptions.NotCorrectAgeException;
+import core.basesyntax.exceptions.NotCorrectPassword;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        nullCheck(user);
+        correctPasswordCheck(user);
+        ageCheck(user);
+        sameLoginCheck(user);
+        storageDao.add(user);
+        return user;
+    }
+
+    private void correctPasswordCheck(User user) {
+        if (user.getPassword().length() <= 6) {
+            throw new NotCorrectPassword("Password to short");
+        }
+    }
+
+    private void sameLoginCheck(User user) {
+        User login = storageDao.get(user.getLogin());
+        if (login != null) {
+            throw new DuplicateAnExistingLoginException("This email "
+                    + user.getLogin() + " has already been taken");
+        }
+
+    }
+
+    private void ageCheck(User user) {
+        if (user.getAge() < 18) {
+            throw new NotCorrectAgeException("Ure are to young");
+        }
+    }
+
+    private void nullCheck(User user) {
+        if (user.getLogin() == null || user.getPassword() == null) {
```

Comment:
```suggestion
        if (user == null || user.getAge() == null || user.getLogin() == null || user.getPassword() == null) {
```

lets also add check for user == null and user.getAge() == null (and add tests for this cases)